python/ray/tune/trainable.py

```diff
@@ -281,25 +281,26 @@ def save_to_object(self):
         """
 
         tmpdir = tempfile.mkdtemp("save_to_object", dir=self.logdir)
-        checkpoint_prefix = self.save(tmpdir)
+        checkpoint_path = self.save(tmpdir)
 
+        # Save all files in subtree.
         data = {}
-        base_dir = os.path.dirname(checkpoint_prefix)
-        for path in os.listdir(base_dir):
-            path = os.path.join(base_dir, path)
-            if path.startswith(checkpoint_prefix):
+        for basedir, _, file_names in os.walk(tmpdir):
+            for file_name in file_names:
+                path = os.path.join(basedir, file_name)
+
                 with open(path, "rb") as f:
-                    data[os.path.basename(path)] = f.read()
+                    data[os.path.relpath(path, tmpdir)] = f.read()
 
         out = io.BytesIO()
         data_dict = pickle.dumps({
-            "checkpoint_name": os.path.basename(checkpoint_prefix),
+            "savedir": tmpdir,
+            "checkpoint_path": checkpoint_path,
```

Comment:
Why is there savedir vs checkpoint_path? I would expect there to be just one checkpoint directory.

Reply:
`save` returns a checkpoint_path, and `restore` takes in the returned
checkpoint_path. I didn't want to change the `save` and `restore` functionality in this PR.

Reply:
Can we add a note that setting this is deprecated?